tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -160,27 +309,54 @@ def _retrieve_slurm_compute_nodes_from_config(remote_command_executor):
     return [node for node in slurm_nodes.splitlines() if "dummy-compute" not in node]
 
 
-def _retrieve_slurm_nodes_from_config(remote_command_executor):
-    retrieve_nodes_command = "sudo grep 'NodeName' /opt/slurm/etc/slurm_parallelcluster_nodes.conf"
+def _retrieve_slurm_nodes_from_config(remote_command_executor, gres=False):
+    if gres:
+        retrieve_nodes_command = "sudo grep 'NodeName' /opt/slurm/etc/slurm_parallelcluster_gres.conf"
```

Comment:
I'd use python regex / contains() instead of grep. This is because when it fails, it's much easier to debug, since the error message contains the full output.

Reply:
Addressed, doing a `cat` file and handling all string logic in python